pkg/api/v1/types.go

```diff
@@ -1981,34 +1981,89 @@ type SecretList struct {
 	Items []Secret `json:"items" description:"items is a list of secret objects; see http://releases.k8s.io/HEAD/docs/user-guide/secrets.md"`
 }
 
-// Type and constants for component health validation.
+type ComponentType string
+
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler"
+)
+
+type ComponentSpec struct {
+	Type    ComponentType `json:"type" description:"type of the component"`
+	Address string        `json:"address,omitempty" description:"address to reach the component (scheme, host, port, path); address paths determined by component type"`
+}
+
+type ComponentPhase string
+
+const (
+	ComponentPending    ComponentPhase = "Pending"
+	ComponentRunning    ComponentPhase = "Running"
+	ComponentTerminated ComponentPhase = "Terminated"
+)
+
 type ComponentConditionType string
 
-// These are the valid conditions for the component.
 const (
-	ComponentHealthy ComponentConditionType = "Healthy"
+	ComponentRunningHealthy    ComponentConditionType = "Healthy"
```

Comment:
The term "healthy" is ambiguous and has no obvious relationship to existing concepts defined in the system: liveness and readiness. We already run control-plane components on Kubelet. How does "healthy" related to probes configured on those instances?

It sounds like you want "Ready", given (3).